File: 1_7/1_7.py
S = "1001"
x = 1
c = S.count('1')
flag = True
N = len(S)
limit = x * c + 1
max_count = 0
count = 0
for i in range(N - 1):
    if S[i] == '0':
        count += 1
        if max_count < count:
            max_count = count
        else:
            count = 0
    if max_count >= limit:
        flag = False
        break
if flag:
    print("Yes")
else:
    print("No")

File: 1_7/1_7.hs
import Data.List

main :: IO ()
main = do
  let s = "1001"
      x = 1
      c = length $ filter (== '1') s
      flag = True
      n = length s
      limit = x * c + 1
      (maxCount, _) = foldl' (\(maxC, count) char -> 
                                if char == '0' 
                                then let newCount = count + 1 
                                     in (max maxC newCount, newCount) 
                                else (maxC, 0)) 
                              (0, 0) 
                              (take (n - 1) s)
      result = if maxCount >= limit then "No" else "Yes"
  putStrLn result


